hydra-filters/src/main/java/com/addthis/hydra/data/filter/value/AbstractMatchStringFilter.java

```diff
@@ -258,6 +268,19 @@ public boolean passedFind(String sv) {
         }
     }
 
+    @Override public void postDecode() {
+        if (refreshMinutes == 0) {
+            postDecodeHelper();
+        } else {
```

Comment:
I noticed a pretty serious issue here - in the case where refresh is enabled, we only call postDecodeHelper in the scheduled executor service. While it's true that we call it immediately, it's called asynchronously without blocking. We need to block here so that this value filter can't be used until the post decode step has occured at least once. Therefore, we should always call postDecodeHelper in this method, and change the scheduled executor to not refresh immediately.

Reply:
Done